maven/src/test/java/org/commonjava/maven/galley/maven/model/view/MavenPomViewTest.java

```diff
@@ -387,4 +387,28 @@ public void repositoriesWithNonProfiles()
         assertFalse( repoId.contains( "profile.repository" ) );
         assertEquals( 1, repoId.size() );
     }
+
+    @Test
+    public void repositoriesWithUrlPropertyInProfile()
```

Comment:
we need a variant of this test that retrieves a dependency from the main POM (*not* from a profile), and gets a version that contains an expression in it...and does that *after* looking at the url in the profile repository. The property should have one value in the main pom, and a different value in the profile. The assertion should be that the dependency version has the value from the main pom. So, the sequence will be:

1. Read the POM -> MavenPomView
2. Get the repository that's declared in the profile -> RepositoryView
3. Get the URL from that repository -> assert it is resolved using values from the profile
4. Get the dependency declared in the main body of the POM (not from a profile)
5. Get the version of the dependency, which is declared with an expression referencing a property that is defined one way in the main pom.xml, and another way in the profile
6. Assert the dependency version uses the value from the main pom.xml

This should show the cross-pollution of profile information I was talking about above, in your changes for MavenPomView.